packages/gatsby/src/bootstrap/schema-hot-reloader.js

```diff
@@ -5,11 +5,10 @@ const { haveEqualFields } = require(`../schema/infer/inference-metadata`)
 const { updateStateAndRunQueries } = require(`../query/query-watcher`)
 const report = require(`gatsby-cli/lib/reporter`)
 
-const inferredTypesChanged = (inferenceMetadata, prevInferenceMetadata) =>
```

Comment:
I realize you only changed the var name but why not use `.some` here?

```
Object.keys(inferenceMetadata).some(type => inferenceMetadata[type].dirty &&	!haveEqualFields(inferenceMetadata[type], prevInferenceMetadata[type]))
```
Would also be faster as it's shortcuts on the first hit.
(If it's more likely not to hit then using an inverted .every is more efficient)

Reply:
Good catch, thanks! That's because originally this function returned a list of `dirty` types and not just boolean. Will change.

Reply:
If the occurrence of one dirty type is more frequent than none at all then we should use `.some` here :)